src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -95,7 +95,9 @@ async function initializeConnection(options: CircuitStartOptions, logger: Logger
   const connection = connectionBuilder.build();
 
   setEventDispatcher((descriptor, args) => {
-    connection.send('DispatchBrowserEvent', JSON.stringify(descriptor), JSON.stringify(args));
+    const getJsonBytes = (value) => new TextEncoder().encode(JSON.stringify(value));
```

Comment:
A long time ago [we added a polyfill for `TextEncoder` (the decoding part)](https://github.com/dotnet/blazor/pull/1168), which is still there now. At the time this was to help with legacy Edge support. Using `TextEncoder` directly here means taking an opinion that we don't need this any more.

### Suggestion

Either:

 * We should remove our `TextEncoder` polyfill (`Utf8Decoder.ts`) - my preferred options
 * Or, we should also polyfill the encoder here.

No problem if we address this in a follow-up PR as there's nothing urgent about it and no reason why @pranavkm should be responsible for it more than anyone else.

### Side-question

In 5.0 [we announced that Blazor Server and WebAssembly would no longer be *fully functional* in Legacy Edge](https://docs.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/5.0/blazor-browser-support-updated#reason-for-change), but I don't see any clear statement about what *fully functional* means. Does it mean it works with polyfills? Or that it doesn't work at all under any circumstances? I don't know. @danroth27 Do you know what expectations we've been setting more recently with customers? Is there any reason to believe any customers are still reliant on polyfilling Legacy Edge support and are we saying anything about that in 6.0?

Reply:
> * We should remove our `TextEncoder` polyfill (`Utf8Decoder.ts`) - my preferred option

I agree with you, we should just remove this.

WRT to 5.0 our statement has never been clear, but I think it's fair to summarize it as "your mileage will vary" as in we think it would work but we don't guarantee/test that it does.